src/taskstatus.js

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable linebreak-style */
+import ToDoList from './todolist.js';
+
+const tdlGlobal = new ToDoList();
+
+export default class taskStatus {
+  constructor() {
+    this.comp = false;
+  }
+
+  update = () => {
+    document.querySelector('div').addEventListener('change', (pressed) => {
+      if (pressed.target.classList.contains('checker')) {
+        this.comp = false;
+        if (pressed.target.checked) {
+          this.comp = true;
+          pressed.target.nextElementSibling.style.textDecoration = 'line-through';
+          pressed.target.nextElementSibling.style.color = '#aaa';
+        } else {
+          pressed.target.nextElementSibling.style.textDecoration = 'none';
+          pressed.target.nextElementSibling.style.color = '#000';
+        }
+        const taskChecker = document.querySelectorAll('.checker');
+        let idx = 0;
+        // eslint-disable-next-line no-restricted-syntax
+        for (const checker of taskChecker) {
+          if (checker.id === pressed.target.id) {
+            break;
+          }
+          idx += 1;
+        }
+        Object.values(tdlGlobal.todolist).forEach((val, index) => {
+          if (idx === index) {
+            val.completed = this.comp;
+          }
+        });
```

Comment:
The example I referred to using the forEach() function to get the index. You use it at this point.

Reply:
Seen